src/caching_key_wrapper.rs

```diff
@@ -229,27 +227,24 @@ where
 
         Ok(())
     }
+}
 
-    /// Get a cached data key or generate one for a certain number of bytes
-    ///
-    /// Note: the bytes field is used to determine when keys should be expired. It's important that
-    /// this is the number of bytes the key will be used to encrypt.
-    pub async fn get_or_generate_data_key_for_bytes(
-        &self,
-        bytes: usize,
-    ) -> Result<DataKey, KeyGenerationError> {
+#[async_trait(?Send)]
+impl<K> KeyProvider for CachingKeyWrapper<K>
+where K: KeyProvider {
+    async fn generate_data_key(&self, bytes: usize) -> Result<DataKey, KeyGenerationError> {
```

Comment:
Is `bytes` the size of the key? Maybe some docs wouldn't go astray?

Reply:
Oof great point 👍🏻